apps/gpc-verifier/app/util/verifyProof.ts

```diff
@@ -0,0 +1,107 @@
+import { Dispatch } from "react";
+import {
+  GPCBoundConfig,
+  GPCProof,
+  gpcArtifactDownloadURL,
+  gpcVerify,
+  deserializeGPCBoundConfig,
+  deserializeGPCRevealedClaims,
+  PODMembershipLists
+} from "@pcd/gpc";
+import { tryRecordNullifierHash } from "@/util/persistence";
+
+export const verifyProof = async (
+  boundConfig: GPCBoundConfig,
+  membershipLists: PODMembershipLists,
+  externalNullifier: string,
+  watermark: string,
+  proofStr: string,
+  setVerified: Dispatch<boolean>
```

Comment:
As I mention in the prover, I think we should split the core logic from the React logic and from the serialization/deserialization.